channel_test.go

```diff
@@ -0,0 +1,77 @@
+package p9p
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+	"time"
+)
+
+type fakeAddr struct{}
+
+func (a fakeAddr) Network() string {
+	return ""
+}
+func (a fakeAddr) String() string {
+	return "fake address"
+}
+
+type writeConnMock struct {
+	data []byte
+}
+
+func (c *writeConnMock) Read(b []byte) (n int, err error) {
+	return 0, errors.New("not implemented")
+}
+
+func (c *writeConnMock) Write(b []byte) (n int, err error) {
+	c.data = append(c.data, b...)
+
+	n = len(b)
+	return n, nil
+}
+
+func (c *writeConnMock) Close() error {
+	return nil
+}
+
+func (c *writeConnMock) LocalAddr() net.Addr {
+	return fakeAddr{}
+}
+
+func (c *writeConnMock) RemoteAddr() net.Addr {
+	return fakeAddr{}
+}
+
+func (c *writeConnMock) SetDeadline(t time.Time) error {
+	return nil
+}
+
+func (c *writeConnMock) SetReadDeadline(t time.Time) error {
+	return nil
+}
+
+func (c *writeConnMock) SetWriteDeadline(t time.Time) error {
+	return nil
+}
+
+func TestReadOverflow(t *testing.T) {
+	const testMsize = 500
```

Comment:
At which point ? when creating the channel ? I am not sure what tells the protocol, but I think having a minimum msize of [twrite header size] + 1 would not hurt.